src/main/dynamo/GSI.ts

```diff
@@ -3,13 +3,15 @@ import { Model } from "./Model"
 import { Table } from "./Table"
 import { ExtractFields } from "./types"
 
+type StringKey<T> = keyof ExtractFields<T> & string
```

Comment:
This tells TypeScript that our object's keys must be strings rather than string | number.